rct229/rules/section5/section5rule28.py

```diff
@@ -0,0 +1,169 @@
+
+    def __init__(self):
+        super(Section5Rule28, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule28.BuildingRule(),
+            index_rmr="baseline",
+            id="5-17",
+            description="Opaque surfaces that are not regulated (not part of opaque building envelope) must be modeled the same in the baseline as in the proposed design. ",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_baseline = context.baseline
+        return {"climate_zone": rmr_baseline["weather"]["climate_zone"]}
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule28.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={},
+                each_rule=Section5Rule28.BuildingRule.UnregulatedSurfaceRule(),
+                index_rmr="baseline",
+            )
+
+        def create_data(self, context, data=None):
+            building = context.baseline
+            # Merge into the existing data dict
+            return {
+                **data,
+                "surface_conditioning_category_dict": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building
+                ),
+            }
+
+        def create_context_list(self, context, data=None):
+            # List of all baseline unregulated surfaces to become the context for Unregulated Surfaces
+            scc = data["surface_conditioning_category_dict"]
+
+            baseline_surfaces = find_all("$..surfaces[*]", context.baseline)
+            proposed_surfaces = find_all("$..surfaces[*]", context.proposed)
+
+            # This assumes that the surfaces matched by IDs between proposed and baseline
+            matched_proposed_surfaces = match_lists_by_id(
+                baseline_surfaces, proposed_surfaces
+            )
+
+            proposed_baseline_surface_pairs = zip(
+                baseline_surfaces, matched_proposed_surfaces
+            )
+
+            return [
+                UserBaselineProposedVals(None, surface_b, surface_p)
+                for surface_b, surface_p in proposed_baseline_surface_pairs
+                if scc[surface_b["id"]] == SCC.UNREGULATED
+            ]
+
+        class UnregulatedSurfaceRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(
+                    Section5Rule28.BuildingRule.UnregulatedSurfaceRule, self
+                ).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    required_fields={},
+                )
+
+            def create_context_list(self, context, data=None):
```

Comment:
In this case, we do not need to override `create_context_list`. Instead, we can simply include the field
```
list_path="subsurfaces[*]"
```
in the initializer method. We needed to override `create_context_list` in BuildingRule because we also wanted to restrict the surfaces to being UNREGULATED.